cpp/src/arrow/array-test.cc

```diff
@@ -1154,6 +1154,54 @@ TEST_F(TestBinaryBuilder, TestScalarAppend) {
     }
   }
 }
+ 
+TEST_F(TestBinaryBuilder, TestCapacityReserve) {
+  vector<string> strings = {"a", "bb", "cc", "ddddd", "eeeee"};
+  int64_t N = static_cast<int>(strings.size());
+  int64_t length = 0;
+  int64_t data_length = 0;
+  int64_t capacity = N;
+ 
+  ASSERT_OK(builder_->Reserve(capacity));
+  ASSERT_OK(builder_->ReserveData(capacity));
+  
+  ASSERT_EQ(builder_->length(), length);
```

Comment:
In googletest, the 1st parameter passed to `ASSERT_EQ` should be the expected result, so flip the argument order here and below